plugins/by-name/lint/default.nix

```diff
@@ -86,113 +84,49 @@ let
     };
 
     parser = {
-      type = lib.types.strLuaFn;
+      type = strLuaFn;
       description = "The code for your parser function.";
       example = ''
         require('lint.parser').from_pattern(pattern, groups, severity_map, defaults, opts)
       '';
-      mandatory = true;
     };
   };
 
-  mkLinterOpts =
-    noDefaults:
-    types.submodule {
-      freeformType = types.attrs;
-
-      options = mapAttrs (
-        optionName:
-        (
-          {
-            mandatory ? false,
-            apply ? x: x,
-            example ? null,
-            type,
-            description,
-          }:
-          mkOption (
-            {
-              inherit apply description example;
-            }
-            // (
-              if
-                noDefaults && mandatory
-              # Make this option mandatory
-              then
-                { inherit type; }
-              # make it optional
-              else
-                {
-                  type = types.nullOr type;
-                  default = null;
-                }
-            )
-          )
-        )
-      ) linterOptions;
-    };
+  mkLinterOpts = types.submodule {
```

Comment:
If this is no longer a factory-function, it should not be named `mk*`. As this is a concrete submodule type, maybe something like `linterType` or `linterOptsType` is more appropriate?